### PR TITLE
[loki-simple-scalable] replace PR #979

### PR DESCRIPTION
**Pull request by zanhsieh**
This replacement for PR #979 since unable to get DCO sign-off.

Original quote:
----
Adding hpa resources for read/write loki-simple-scaling services. read looks like it was intended to have this but wasnt finished (values with no template files) and write didnt have either.

I think this may need a warning about enabling autoscaling on writes when not using an object_store backend.

### CODE REVIEW

File: charts/loki-simple-scalable/README.md [hunk 1]
```diff
@@ -80,9 +80,9 @@ helm repo add grafana https://grafana.github.io/helm-charts
 | global.image.registry | string | `nil` | Overrides the Docker registry globally for all images |
 | global.priorityClassName | string | `nil` | Overrides the priorityClassName for all pods |
 | imagePullSecrets | list | `[]` | Image pull secrets for Docker images |
-| loki.commonConfig | object | `{"path_prefix":"/var/loki","replication_factor":1,"storage":{"filesystem":{"chunks_directory":"/var/loki/chunks","rules_directory":"/var/loki/rules"}}}` | Check https://grafana.com/docs/loki/latest/configuration/#common_config for more info on how to provide a common configuration |
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/README.md:83]: we don't want to lose this section

File: charts/loki-simple-scalable/values.yaml [hunk 2]
```diff
@@ -59,68 +59,53 @@ loki:
   existingSecretForConfig: ""
   # -- Config file contents for Loki
   # @default -- See values.yaml
-  config: |
+  config:
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/values.yaml:62]: this should be a string

File: charts/loki-simple-scalable/values.yaml [hunk 3]
```diff
@@ -59,68 +59,53 @@ loki:
...
       http_listen_port: 3100
 
     memberlist:
       join_members:
-        - {{ include "loki.fullname" . }}-memberlist
+        - "{{ include \"loki.fullname\" . }}-memberlist"
 
-    {{- if .Values.loki.commonConfig}}
     common:
-    {{- toYaml .Values.loki.commonConfig | nindent 2}}
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/values.yaml:74]: don't want to lose this either

File: charts/loki-simple-scalable/values.yaml [hunk 4]
```diff
@@ -188,6 +173,17 @@ prometheusRule:
 write:
   # -- Number of replicas for the write
   replicas: 1
+  autoscaling:
+    # -- Enable autoscaling for the write
+    enabled: false
+    # -- Minimum autoscaling replicas for the write
+    minReplicas: 1
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/values.yaml:180]: let's set minReplicas to 3 so it will work with the default RF of 3

**Review by trevorwhitney — CHANGES_REQUESTED** [review]
this looks like it's pretty out of date with the current state of main. we lost a bunch of changes, like making the config a string and extracting the common config section

File: charts/loki-simple-scalable/values.yaml (hunk 2 above)
Comment (zanhsieh) [reply charts/loki-simple-scalable/values.yaml:62]: this one won't fix due to line 63 auth_enabled: false is not string.

File: charts/loki-simple-scalable/values.yaml (hunk 3 above)
Comment (zanhsieh) [reply charts/loki-simple-scalable/values.yaml:74]: this one have dependencies and the value has been copied over. won't fix.

File: charts/loki-simple-scalable/README.md (hunk 1 above)
Comment (zanhsieh) [reply charts/loki-simple-scalable/README.md:83]: this part has been moved to under loki.config.

**Comment by zanhsieh** [comment]
@trevorwhitney
Could you approve this PR please? Or you have some idea?

File: charts/loki-simple-scalable/templates/configmap.yaml [hunk 5]
```diff
@@ -7,5 +7,5 @@ metadata:
     {{- include "loki.labels" . | nindent 4 }}
 data:
   config.yaml: |
-    {{- tpl (mergeOverwrite (tpl .Values.loki.config . | fromYaml) .Values.loki.structuredConfig | toYaml) . | nindent 4 }}
+    {{- tpl (toYaml .Values.loki.config) . | nindent 4 }}
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/templates/configmap.yaml:10]: this is going to effectively remove the `structuredConfig` section, is that desired?

File: charts/loki-simple-scalable/templates/read/statefulset-read.yaml [hunk 6]
```diff
@@ -74,8 +76,6 @@ spec:
           volumeMounts:
             - name: config
               mountPath: /etc/loki/config
-            - name: tmp
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/templates/read/statefulset-read.yaml:77]: why are we removing the `/tmp` volume? I'm not actually sure what it's used for, but I can't tell from this PR why it's being removed either.

File: charts/loki-simple-scalable/values.yaml (hunk 2 above)
Comment (trevorwhitney) [reply charts/loki-simple-scalable/values.yaml:62]: we are moving in the direction of string config fields for all the charts. I don't understand what you mean about line 63, and us not being able to include `auth_enabed: false` in the config string?

without this fix this PR cannot be merged.

File: charts/loki-simple-scalable/values.yaml (hunk 3 above)
Comment (trevorwhitney) [reply charts/loki-simple-scalable/values.yaml:74]: again I don't understand. what dependencies prevent us from having a common config section?

File: charts/loki-simple-scalable/values.yaml [hunk 7]
```diff
@@ -59,68 +59,54 @@ loki:
...
   storageConfig: {}
-  # -- Uncomment to configure each storage individually
-  # boltdb_shipper: {}
-  # filesystem: {}
-  # azure: {}
-  # gcs: {}
-  # s3: {}
-
-  # -- Structured loki configuration, takes precedence over `loki.config`, `loki.schemaConfig`, `loki.storageConfig`
-  structuredConfig: {}
```
Comment (trevorwhitney) [inline charts/loki-simple-scalable/values.yaml:123]: again, why are we removing this.

**Review by trevorwhitney — CHANGES_REQUESTED** [review]
This PR is diverging pretty strongly from some common patterns we have across our helm charts, most notably being using strings for the config field. There needs to be a much wider discussion before we make the following changes:

1. Remove the structured config
2. Change the config from a string to a structured value
3. Remove the common config section

If the goal is to get autoscaling merged, I would split this PR to only include those sections, as that could be merged.

**Pull request closed without merging**